scipy/stats/tests/test_stats.py

```diff
@@ -2756,29 +2756,34 @@ def test_sem_nan_policy(self, xp):
         assert_raises(ValueError, stats.sem, x, nan_policy='foobar')
 
 
+@array_api_compatible
+@pytest.mark.usefixtures("skip_xp_backends")
 class TestZmapZscore:
 
     @pytest.mark.parametrize(
         'x, y',
         [([1, 2, 3, 4], [1, 2, 3, 4]),
          ([1, 2, 3], [0, 1, 2, 3, 4])]
     )
-    def test_zmap(self, x, y):
-        z = stats.zmap(x, y)
+    def test_zmap(self, x, y, xp):
         # For these simple cases, calculate the expected result directly
         # by using the formula for the z-score.
-        expected = (x - np.mean(y))/np.std(y)
-        assert_allclose(z, expected, rtol=1e-12)
+        expected = (x - np.mean(y)) / np.std(y)
 
-    def test_zmap_axis(self):
+        x, y = xp.asarray(x), xp.asarray(y)
+        z = stats.zmap(x, y)
+        default_float = xp.asarray(1.).dtype
+        xp_assert_close(z, xp.asarray(expected, dtype=default_float))
```

Comment:
Is there a reason not to compute `expected` using `xp` to avoid the dtype dance?

Reply:
Not sure what I was thinking there. I'll try that.

Reply:
Had to change inputs to floats and use `xp_test = array_namespace(x, y)` to get `std` to have `correction`. But I changed it.